app/controllers/shipping_api_controller.rb

```diff
@@ -17,20 +23,41 @@ def create_log
 
   private
 
-  def set_origin
+  def origin
+    ShippingLocation.new(origin_params)
+  end
+
+  def origin_params
+    params.require(:origin).permit(:city, :state, :zip, :country)
+  end
+
+  def destination
+    ShippingLocation.new(destination_params)
+  end
+
+  def destination_params
+    params.require(:destination).permit(:city, :state, :zip, :country)
   end
 
-  def set_destination
+  def package
+    weight = package_params[:weight].to_i
+    dimensions = package_params[:dimensions].map{|d| d.to_i}
```

Comment:
Is `to_i` necessary on weight and dimensions? Or can the gem handle these as strings?